### ISSUE TITLE
Idris doesn't recognise when .ibc is from other (git) version

### ISSUE THREAD

**Issue by Garnasha** [idris-lang/Idris-dev#4356]
Multiple Idris versions seem unable to deal with .ibc files left behind by any versions except themselves. Haven't tried to reproduce with different x.y.z version numbers yet, but 1.2.0 and 1.2.0-git:cc7ef9ad certainly conflict.

# Steps to Reproduce
Install two versions of Idris on your system. I used the one from the Arch package (version 1.2.0) and one built by stack from cc7ef9ad (version 1.2.0-git:cc7ef9ad) naming the latter idris-stack to disambiguate them. Take any source file, and open it in the REPL (either as command-line argument or :load works) with one version. Close the REPL, and open it with the other version.

# Expected Behavior
Both versions do their own compilation, and can peacefully coexist, possibly at the cost of having to recompile every time after another version has run. 
```
[02:18 sal@tryptophan ~/edu/courses/compiler_construction]$ idris TestFrame.idr
     ____    __     _                                          
    /  _/___/ /____(_)____                                     
    / // __  / ___/ / ___/     Version 1.2.0
  _/ // /_/ / /  / (__  )      http://www.idris-lang.org/      
 /___/\__,_/_/  /_/____/       Type :? for help               

Idris is free software with ABSOLUTELY NO WARRANTY.            
For details type :warranty.
Type checking ./TestFrame.idr
*TestFrame> 
[02:20 sal@tryptophan ~/edu/courses/compiler_construction]$ idris-stack TestFrame.idr
     ____    __     _                                          
    /  _/___/ /____(_)____                                     
    / // __  / ___/ / ___/     Version 1.2.0-git:cc7ef9ad
  _/ // /_/ / /  / (__  )      http://www.idris-lang.org/      
 /___/\__,_/_/  /_/____/       Type :? for help               

Idris is free software with ABSOLUTELY NO WARRANTY.            
For details type :warranty.
Type checking ./TestFrame.idr
*TestFrame> 
```

# Observed Behavior
The first version to read the file is the only one that can read it until either the file access time changes, or the .ibc is deleted. The other one crashes with a Haskell CallStack report.
```
[02:21 sal@tryptophan ~/edu/courses/compiler_construction]$ idris TestFrame.idr 
     ____    __     _                                          
    /  _/___/ /____(_)____                                     
    / // __  / ___/ / ___/     Version 1.2.0
  _/ // /_/ / /  / (__  )      http://www.idris-lang.org/      
 /___/\__,_/_/  /_/____/       Type :? for help               

Idris is free software with ABSOLUTELY NO WARRANTY.            
For details type :warranty.
Type checking ./TestFrame.idr
*TestFrame> 
Bye bye
[02:21 sal@tryptophan ~/edu/courses/compiler_construction]$ idris-stack TestFrame.idr 
     ____    __     _                                          
    /  _/___/ /____(_)____                                     
    / // __  / ___/ / ___/     Version 1.2.0-git:cc7ef9ad
  _/ // /_/ / /  / (__  )      http://www.idris-lang.org/      
 /___/\__,_/_/  /_/____/       Type :? for help               

Idris is free software with ABSOLUTELY NO WARRANTY.            
For details type :warranty.
idris-stack: Data.Binary.Get.runGet at position 55: Unknown encoding for constructor
CallStack (from HasCallStack):
  error, called at libraries/binary/src/Data/Binary/Get.hs:351:5 in binary-0.8.5.1:Data.Binary.Get
[02:22 sal@tryptophan ~/edu/courses/compiler_construction]$ 
```
Switching the roles around doesn't change behaviour except for the position reported in runGet
```
<snip>
Version 1.2.0
<snip>
idris: Data.Binary.Get.runGet at position 28453: Unknown encoding for constructor
CallStack (from HasCallStack):
  error, called at libraries/binary/src/Data/Binary/Get.hs:351:5 in binary-0.8.5.1:Data.Binary.Get
```

Using `:load` also fails, same error, but doesn't crash the REPL.
```
Idris> :load TestFrame.idr 
Data.Binary.Get.runGet at position 28453: Unknown encoding for constructor
CallStack (from HasCallStack):
  error, called at libraries/binary/src/Data/Binary/Get.hs:351:5 in binary-0.8.5.1:Data.Binary.Get
Idris> 
```

**Comment by Garnasha** [comment]
This is probably solved by 507a731 . Question: should the IBC version be based on the Idris version, or should it be manually bumped only when something relevant to it changes? In the latter case, should Idris be able to gracefully deal with cases where when the IBC version should be different but isn't (for instance, catching the error and recompiling the offending file)?

**Comment by ahmadsalim** [comment]
@Garnasha Thanks for reporting the issue.
Yes, this is a developer mistake as @david-christiansen mentioned. I will close the issue.
